frontend/src/Frontend/JsonData.hs

```diff
@@ -306,11 +318,14 @@ makeKeysets walletL cfg =
       let
         keys :: Dynamic t (Map KeyName PublicKey)
         keys = do
-          names <- Set.toList <$> keynames
-          let namesMap = Map.fromList . map (,()) $ names
-          wKeys <- walletL ^. wallet_keys
-          let keyPairs = Map.intersection wKeys namesMap
-          pure $ _keyPair_publicKey <$> keyPairs
+          names <- keynames
+          wAccounts <- walletL ^. wallet_accounts
+          -- This isn't great... should really come back to this
```

Comment:
Fold over `IntMap.elems wAccounts` ? That would let you do the filtering and building of the new map in a single pass?

Reply:
I converted it to a fold, it's a little neater. I can't remember exactly what I was unhappy about.